templates/app/routes/_testing/index.js

```diff
@@ -0,0 +1,17 @@
+'use strict'
+const { join } = require('path')
+const autoload = require('fastify-autoload')
+
+// use this file to create a plugin that provides reusable state to all routes
+// this is useful for isolating routes by decoupling from root app.js and for
+// creating reusable resource pools, such as a database connection
```

Comment:
I'm not really ok with this approach. This will lead to global state being managed in a plugin, which defeats the arguments for plugins.

IMHO, fastify-test should support some ways of:

1. configuring the options for the plugins to pass in a connection pool
2. a way to create said  connection pool before the first test is executed and destroyed after the last.